app/src/main/java/io/homeassistant/companion/android/sensors/CarSensorManager.kt

```diff
@@ -0,0 +1,296 @@
+            "mdi:barrel",
+            unitOfMeasurement = "%",
+            stateClass = SensorManager.STATE_CLASS_MEASUREMENT,
+            deviceClass = "battery"
+        )
+        private val batteryLevel = SensorManager.BasicSensor(
+            "car_battery_level",
+            "sensor",
+            R.string.basic_sensor_name_car_battery_level,
+            R.string.sensor_description_car_battery_level,
+            "mdi:car-battery",
+            unitOfMeasurement = "%",
+            stateClass = SensorManager.STATE_CLASS_MEASUREMENT,
+            deviceClass = "battery",
+            entityCategory = SensorManager.ENTITY_CATEGORY_DIAGNOSTIC
+        )
+        private val carName = SensorManager.BasicSensor(
+            "car_name",
+            "sensor",
+            R.string.basic_sensor_name_car_name,
+            R.string.sensor_description_car_name,
+            "mdi:car-info"
+        )
+        private val carStatus = SensorManager.BasicSensor(
+            "car_charging_status",
+            "sensor",
+            R.string.basic_sensor_name_car_charging_status,
+            R.string.sensor_description_car_charging_status,
+            "mdi:ev-station",
+            deviceClass = "plug"
+        )
+        private val odometerValue = SensorManager.BasicSensor(
+            "car_odometer",
+            "sensor",
+            R.string.basic_sensor_name_car_odometer,
+            R.string.sensor_description_car_odometer,
+            "mdi:map-marker-distance",
+            unitOfMeasurement = "m",
+            stateClass = SensorManager.STATE_CLASS_TOTAL_INCREASING,
+            deviceClass = "distance"
+        )
+
+        private val sensorsList = listOf(
+            batteryLevel,
+            carName,
+            carStatus,
+            fuelLevel,
+            odometerValue
+        )
+
+        // track if we have already sent the "open app" message for each sensor
+        private val connectList = mutableMapOf(
+            batteryLevel to false,
+            carName to false,
+            carStatus to false,
+            fuelLevel to false,
+            odometerValue to false
+        )
+
+        private enum class Listener {
+            ENERGY, MODEL, MILEAGE, STATUS,
+        }
+
+        private val listenerSensors = mapOf(
+            Listener.ENERGY to listOf(batteryLevel, fuelLevel),
+            Listener.MODEL to listOf(carName),
+            Listener.STATUS to listOf(carStatus),
+            Listener.MILEAGE to listOf(odometerValue)
+        )
+        private val listenerLastRegistered = mutableMapOf(
+            Listener.ENERGY to -1L,
+            Listener.MODEL to -1L,
+            Listener.STATUS to -1L,
+            Listener.MILEAGE to -1L
+        )
+    }
+
+    override val name: Int
+        get() = R.string.sensor_name_car
+
+    override suspend fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
According to the [docs](https://developer.android.com/training/cars/apps#carinfo) you linked earlier, we need to hide the odometer sensor from automotive builds.

![image](https://github.com/home-assistant/android/assets/1634145/f7bea4bf-0336-46e8-b49a-6bd37f2f0a94)

In fact if you don't want to refactor `requiredPermissions` in this PR (since its already getting pretty big), we can hide these sensors from the automotive build for now.  That may be better too for now as we have some sensors in that build that need to be hidden anyways.